src/dlgTriggerEditor.cpp

```diff
@@ -158,8 +159,14 @@ dlgTriggerEditor::dlgTriggerEditor( Host * pH )
     pVB1->addWidget( mpAliasMainArea );
 
     mpActionsMainArea = new dlgActionMainArea( mainArea );
-    mpActionsMainArea->setSizePolicy( sizePolicy8 );
-    pVB1->addWidget( mpActionsMainArea );
+    mpActionsMainArea->setSizePolicy( sizePolicy );
+    pVB1->addWidget( mpActionsMainArea, Qt::AlignTop );
+    connect(mpActionsMainArea->pushButton_changeIcon, SIGNAL(pressed()), this, SLOT(slot_chose_action_icon()));
```

Comment:
Actually the Icon feature is NOT a new feature it was more functionality that got dropped at the same time as the command up/down entry boxes in the "safe mapper" commit 4e651d55fd7a73af24fe66ae3d85e9e5b13a1fc5 which also did away with custom button geometry, flat buttons and coloured buttons - I was not convinced of the flat effect and thought it might interfere with QSS/CSS; getting the custom geometry to work properly is not something I have completely solved as of yet and the button colour stuff uses `QPalette` which is regarded by Qt as incompatible with `QStyleSheets` {QSS/CSS} anyhow.  As for icons not being portable I guess it really ought to be coded to allow relative paths so that the icons can be bundled into a package - with the starting point being the profile home directory.  You do need a separate clear button to allow a non-destructive cancel from the icon file selection dialog, and the display-only QLineEdit for the icon name has always been on the form (just it had got resized to (0,0))!